### DISCUSSION TITLE
[ALG101] - LIOJ 1046

### DISCUSSION THREAD

**Discussion by backas36** [Lidemy/forum#76] [前端相關問題]
這邊我遇到的問題是關於邏輯運算，我想問一下 1046 這個題目如果是這樣去假設邏輯，會有什麼問題？
```js
if( lines[i][0] === lines[i][1]  === lines[i][2]){
      return lines[i][0]
    }
```
我自己想得到可能會遇到的錯誤是：
`lines[i][0] === lines[i][1]  === lines[i][2]` 
1. 依據順序會先從 `lines[i][0] === lines[i][1]` 先判斷，所以程式會變成 ` true === lines[i][2]`
2. 因為使用 === 所以類型也不同，會回傳 false
3. 如果使用 == 一樣是 false，因為 true 會先轉換成 1 之後做比較，變成 ` 1 == lines[i][2]`，所以也會回傳 false。

但是不確定**這樣的理解是否正確或者是還有其他地方我是我沒有聯想到的**？

**Comment by aszx87410** [comment]
理解完全正確，這判斷式在上次直播裡面有提到喔

**Reply by backas36** [reply]
謝謝老師！直播目前只追到 week1 >"< 哈哈 努力中
